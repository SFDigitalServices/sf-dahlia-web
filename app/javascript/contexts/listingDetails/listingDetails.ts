interface ListingDetails {
  /**  */
  uid: string

  /**  */
  email: string

  /**  */
  created_at: Date

  /**  */
  updated_at: Date
}

interface ListingDetailsData {
  data: any
  success: boolean
}

export { ListingDetails, ListingDetailsData }
