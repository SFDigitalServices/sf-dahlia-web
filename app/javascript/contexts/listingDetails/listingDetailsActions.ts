import { createAction } from "typesafe-actions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"

export enum ListingDetailsActions {
  StartFetchingUnits = "SetFetchingUnits",
  FinishFetchingUnits = "FinishFetchingUnits",
  SetUnits = "SetUnits",
  SetAmiCharts = "SetAmiCharts",
  StartFetchingAmiCharts = "SetFetchingAmiCharts",
  SetFetchedAmiCharts = "SetFetchedAmiCharts",
  SetFetchingAmiChartsError = "SetFetchingAmiChartsError",
  SetFetchingUnitsError = "SetFetchingUnitsError",
  FinishFetchingAmiCharts = "FinishFetchingAmiCharts",
}

export const startFetchingUnits = createAction(ListingDetailsActions.StartFetchingUnits)()
export const setUnits = createAction(ListingDetailsActions.SetUnits)()
export const setAmiCharts = createAction(ListingDetailsActions.SetAmiCharts)()
export const finishFetchingAmiCharts = createAction(ListingDetailsActions.FinishFetchingAmiCharts)<
  RailsAmiChart[]
>()
export const setFetchedAmiCharts = createAction(ListingDetailsActions.SetFetchedAmiCharts)()
export const startFetchingAmiCharts = createAction(ListingDetailsActions.StartFetchingAmiCharts)()
export const finishFetchingUnits = createAction(ListingDetailsActions.FinishFetchingUnits)<
  RailsUnit[]
>()
export const setFetchingUnitsError = createAction(
  ListingDetailsActions.SetFetchingUnitsError
)<Error>()
export const setFetchingAmiChartsError = createAction(
  ListingDetailsActions.SetFetchingAmiChartsError
)<Error>()
