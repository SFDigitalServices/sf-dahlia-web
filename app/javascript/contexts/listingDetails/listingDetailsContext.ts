import { createContext } from "react"
import RailsUnit from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"

export type ContextProps = {
  units: RailsUnit[]
  amiCharts: RailsAmiChart[]
  fetchedAmiCharts: boolean
  fetchingAmiCharts: boolean
  fetchedUnits: boolean
  fetchingUnits: boolean
  fetchUnits: (listingId: string) => void
  fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => void
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

export default createContext<Partial<ContextProps>>({})
