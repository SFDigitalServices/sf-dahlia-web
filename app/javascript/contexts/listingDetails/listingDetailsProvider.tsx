import React, { useReducer } from "react"
import ListingDetailsReducer from "./listingDetailsReducer"
import ListingDetailsContext, { ContextProps } from "./listingDetailsContext"
import { getUnits, getAmiCharts } from "../../api/listingApiService"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"
import {
  finishFetchingUnits,
  finishFetchingAmiCharts,
  startFetchingAmiCharts,
  startFetchingUnits,
  setFetchingAmiChartsError,
  setFetchingUnitsError,
} from "./listingDetailsActions"

interface ListingDetailsProviderProps {
  children?: React.ReactNode
}

const ListingDetailsProvider = (props: ListingDetailsProviderProps) => {
  const [state, dispatch] = useReducer(ListingDetailsReducer, {
    units: [],
    amiCharts: [],
    fetchingUnits: false,
    fetchedUnits: false,
    fetchingAmiCharts: false,
    fetchedAmiCharts: false,
    fetchingAmiChartsError: null,
    fetchingUnitsError: null,
  })

  const contextValues: ContextProps = {
    units: state.units,
    amiCharts: state.amiCharts,
    fetchedAmiCharts: state.fetchedAmiCharts,
    fetchingAmiCharts: state.fetchingAmiCharts,
    fetchedUnits: state.fetchedUnits,
    fetchingUnits: state.fetchingUnits,
    fetchingAmiChartsError: state.fetchingAmiChartsError,
    fetchingUnitsError: state.fetchingUnitsError,
    fetchUnits: (listingId: string) => {
      dispatch(startFetchingUnits())
      getUnits(listingId)
        .then((units: RailsUnit[]) => {
          dispatch(finishFetchingUnits(units))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingUnitsError(error))
        })
    },
    fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => {
      dispatch(startFetchingAmiCharts())
      getAmiCharts(chartsToFetch)
        .then((amiCharts: RailsAmiChart[]) => {
          dispatch(finishFetchingAmiCharts(amiCharts))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingAmiChartsError(error))
        })
    },
  }

  return (
    <ListingDetailsContext.Provider value={contextValues}>
      {props.children}
    </ListingDetailsContext.Provider>
  )
}

export default ListingDetailsProvider
