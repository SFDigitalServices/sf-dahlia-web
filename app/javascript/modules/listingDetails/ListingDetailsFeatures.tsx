import React from "react"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsUnitAccordions } from "./ListingDetailsUnitAccordions"
import { AdditionalFees, Description, ListingDetailItem, t } from "@bloom-housing/ui-components"
import { isBMR, isRental, isSale } from "../../util/listingUtil"
import { stripMostTags } from "../../util/filterUtil"
import { isValidUrl } from "../../util/urlUtil"
import ErrorBoundary, { BoundaryScope } from "../../components/ErrorBoundary"
import { getTranslatedString, renderMarkup } from "../../util/languageUtil"
import { RailsTranslations } from "../../api/types/rails/listings/RailsTranslation"

export interface ListingDetailsFeaturesProps {
  listing: RailsListing
  imageSrc: string
}

const getDepositString = (min?: string, max?: string) => {
  if (!min && !max) return null
  if (min && max) return `$${min} - $${max}`
  return min ? `$${min}` : `$${max}`
}

// TODO: add prop for items that should have machine translated content
interface FeatureItemProps {
  content: string
  title: string
  fieldName?: string
  translations?: RailsTranslations
  toTranslate?: boolean
}
const FeatureItem = ({
  content,
  fieldName,
  title,
  translations,
  toTranslate,
}: FeatureItemProps) => {
  if (!content) return <></>
  const description = toTranslate ? getTranslatedString(content, fieldName, translations) : content
  return (
    <Description
      term={title}
      description={stripMostTags(description)}
      markdownProps={{
        disableParsingRawHTML: false,
      }}
      markdown={true}
      dtClassName={toTranslate && "translate"}
    />
  )
}

interface UnitDetailsFeatureItemProps {
  pricingMatrixUrl: string
}
const UnitDetailsFeatureItem = ({ pricingMatrixUrl }: UnitDetailsFeatureItemProps) => {
  const hyperlink = isValidUrl(pricingMatrixUrl)
    ? `<a href='${pricingMatrixUrl}' target="_blank">${t(
        "listings.features.downloadPriceAndIncomeLimitForEachUnitPdf"
      )}</a>`
    : ""

  return (
    <>
      <Description
        term={t("listings.features.unitDetails")}
        description={hyperlink}
        markdownProps={{
          disableParsingRawHTML: false,
        }}
        markdown={true}
      />
      <Description term="" description="" />
    </>
  )
}

export const ListingDetailsFeatures = ({ listing, imageSrc }: ListingDetailsFeaturesProps) => {
  const depositSubtext = [t("listings.features.orOneMonthsRent")]

  if (isBMR(listing)) {
    depositSubtext.push(t("listings.features.mayBeHigherForLowerCreditScores"))
  }

  const translations = listing.translations

  return (
    <ListingDetailItem
      imageAlt={""}
      imageSrc={imageSrc}
      title={t("listings.features.header")}
      subtitle={
        isSale(listing)
          ? t("listings.features.saleSubheader")
          : t("listings.features.rentSubheader")
      }
    >
      <div className="listing-detail-panel">
        <dl className="column-definition-list">
          <FeatureItem
            content={listing.Neighborhood}
            title={t("listings.neighborhood.header")}
            toTranslate={false}
          />
          <FeatureItem
            content={String(listing.Year_Built)}
            title={t("listings.features.built")}
            toTranslate={false}
          />
          <FeatureItem
            content={listing.Appliances}
            title={t("listings.features.appliances")}
            toTranslate={true}
            fieldName={"Appliances__c"}
            translations={translations}
          />
          <FeatureItem
            content={listing.Services_Onsite}
            title={t(
              isSale(listing)
                ? "listings.features.servicesCoveredByHoaDues"
                : "listings.features.servicesOnsite"
            )}
            toTranslate={true}
            fieldName={"Services_Onsite__c"}
            translations={translations}
          />
          <FeatureItem
            content={listing.Parking_Information}
            title={t("listings.features.parking")}
            toTranslate={true}
            fieldName={"Parking_Information__c"}
            translations={translations}
          />
          {isRental(listing) && (
            <FeatureItem
              content={listing.Utilities}
              title={t("listings.features.utilities")}
              toTranslate={true}
              fieldName={"Utilities__c"}
              translations={translations}
            />
          )}
          <FeatureItem
            content={listing.Smoking_Policy}
            title={t("listings.features.smokingPolicy")}
            toTranslate={true}
            fieldName={"Smoking_Policy__c"}
            translations={translations}
          />
          <FeatureItem
            content={listing.Pet_Policy}
            title={t("listings.features.petsPolicy")}
            toTranslate={true}
            fieldName={"Pet_Policy__c"}
            translations={translations}
          />
          <FeatureItem
            content={listing.Amenities}
            title={t("listings.features.propertyAmenities")}
            toTranslate={true}
            fieldName={"Amenities__c"}
            translations={translations}
          />
          <FeatureItem
            content={listing.Accessibility}
            title={t("listings.features.accessibility")}
            toTranslate={true}
            fieldName={"Accessibility__c"}
            translations={translations}
          />

          {isRental(listing) && (
            <Description term={t("listings.features.unitFeatures")} description={""} />
          )}
          {isSale(listing) && <UnitDetailsFeatureItem pricingMatrixUrl={listing.Pricing_Matrix} />}
        </dl>
        <ErrorBoundary boundaryScope={BoundaryScope.component}>
          <ListingDetailsUnitAccordions />
        </ErrorBoundary>
        {isRental(listing) && (
          <AdditionalFees
            deposit={getDepositString(
              listing.Deposit_Min?.toLocaleString(),
              listing.Deposit_Max?.toLocaleString()
            )}
            applicationFee={listing.Fee ? `$${listing.Fee.toFixed(2)?.toLocaleString()}` : null}
            footerContent={[
              <span className="translate">
                {renderMarkup(
                  getTranslatedString(
                    listing?.Costs_Not_Included,
                    "Costs_Not_Included__c",
                    listing?.translations
                  )
                )}
              </span>,
            ]}
            strings={{
              sectionHeader: t("listings.features.additionalFees"),
              deposit: t("listings.features.deposit"),
              depositSubtext: depositSubtext,
              applicationFee: t("listings.features.applicationFee"),
              applicationFeeSubtext: [
                t("listings.features.perApplicant"),
                t("listings.features.duePostLottery"),
              ],
            }}
          />
        )}
      </div>
    </ListingDetailItem>
  )
}
