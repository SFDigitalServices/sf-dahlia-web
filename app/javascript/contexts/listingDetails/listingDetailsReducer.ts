import { createReducer } from "typesafe-actions"
import { ListingDetailsActions } from "./listingDetailsActions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"
import { AmiChartData } from "./listingDetailsContext"

type ListingDetailsState = {
  fetchingUnits: boolean
  fetchedUnits: boolean
  fetchingAmiCharts: boolean
  fetchedAmiCharts: boolean
  units: RailsUnit[]
  amiChartData: AmiChartData
  amiCharts: RailsAmiChart[]
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState, {
  [ListingDetailsActions.StartFetchingUnits]: (state) => {
    return { ...state, fetchingUnits: true }
  },
  [ListingDetailsActions.StartFetchingAmiCharts]: (state) => {
    return { ...state, fetchingAmiCharts: true }
  },
  [ListingDetailsActions.FinishFetchingUnits]: (state, { payload: units }) => {
    const amiChartYears: Array<number> = []
    const amiChartTypes = []
    const amiChartPercentages = []

    units.forEach((unit: RailsUnit) => {
      const amiChartYear = amiChartYears.find((year) => {
        return year === unit.AMI_chart_year
      })

      const amiChartType = amiChartTypes.find((type) => {
        return type === unit.AMI_chart_type
      })

      const amiChartPercentage = amiChartPercentages.find((type) => {
        return type === unit.Max_AMI_for_Qualifying_Unit
      })

      if (!amiChartYear) {
        amiChartYears.push(unit.AMI_chart_year)
      }

      if (!amiChartType) {
        amiChartTypes.push(unit.AMI_chart_type)
      }

      if (!amiChartPercentage) {
        amiChartPercentages.push(unit.Max_AMI_for_Qualifying_Unit)
      }
    })

    return {
      ...state,
      fetchingUnits: false,
      fetchedUnits: true,
      units,
      amiChartData: {
        years: amiChartYears,
        percentages: amiChartPercentages,
        types: amiChartTypes,
      },
    }
  },
  [ListingDetailsActions.FinishFetchingAmiCharts]: (state, { payload: amiCharts }) => {
    return {
      ...state,
      fetchingAmiCharts: false,
      fetchedAmiCharts: true,
      amiCharts: amiCharts?.map((amiChart: RailsAmiChart) => {
        return {
          ...amiChart,
          chartType: amiChart?.values[0]?.chartType,
        }
      }),
    }
  },
})

export default ListingDetailsReducer
