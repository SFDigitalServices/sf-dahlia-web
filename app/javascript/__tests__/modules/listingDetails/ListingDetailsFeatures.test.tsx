import React from "react"
import { render, cleanup } from "@testing-library/react"
import { ListingDetailsFeatures } from "../../../modules/listingDetails/ListingDetailsFeatures"
import { closedRentalListing } from "../../data/RailsRentalListing/listing-rental-closed"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"
import { units } from "../../data/RailsListingUnits/listing-units"
import { preferences as defaultPreferences } from "../../data/RailsListingPreferences/lottery-preferences-default"
import ListingDetailsContext from "../../../contexts/listingDetails/listingDetailsContext"

const axios = require("axios")

jest.mock("axios")

describe("ListingDetailsFeatures", () => {
  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
    jest.resetAllMocks()
  })

  it("displays listing details features section when rental listing", async (done) => {
    // This component pulls in react-media, which needs this custom mock
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })

    const { asFragment, findAllByTestId } = render(
      <ListingDetailsContext.Provider
        value={{
          units: units,
          amiCharts: [],
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: null,
          fetchingUnitsError: null,
        }}
      >
        <ListingDetailsFeatures listing={closedRentalListing} imageSrc={"listing-features.svg"} />
      </ListingDetailsContext.Provider>
    )

    expect(await findAllByTestId("content-accordion-button")).toHaveLength(3)
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("displays listing details features section when sales listing", async (done) => {
    axios.get.mockResolvedValue({
      data: { units: openSaleListing.Units, preferences: defaultPreferences },
    })

    // This component pulls in react-media, which needs this custom mock
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        matches: true,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }
    })

    const { asFragment, findAllByTestId } = render(
      <ListingDetailsContext.Provider
        value={{
          units: units,
          amiCharts: [],
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: null,
          fetchingUnitsError: null,
        }}
      >
        <ListingDetailsFeatures listing={openSaleListing} imageSrc={"listing-features.svg"} />
      </ListingDetailsContext.Provider>
    )

    expect(await findAllByTestId("content-accordion-button")).toHaveLength(3)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
})
