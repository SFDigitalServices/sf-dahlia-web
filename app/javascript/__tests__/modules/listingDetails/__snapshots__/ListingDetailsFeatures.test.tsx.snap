// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsFeatures displays a pdf hyperlink when Pricing_Matrix url exists in sales listing 1`] = `
<DocumentFragment>
  <li
    class="responsive-content-item "
  >
    <header
      class="detail-header "
    >
      <span
        class="detail-header__image-container"
      >
        <img
          alt=""
          class="detail-header__image "
          src="listing-features.svg"
        />
      </span>
      <hgroup
        class="detail-header__hgroup"
      >
        <h2
          class="detail-header__title"
        >
          Features
        </h2>
        <span
          class="detail-header__subtitle"
        >
          Amenities and unit details
        </span>
        <span
          class="ui-icon ui-medium"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </hgroup>
    </header>
    <div
      class="listing-detail-panel"
    >
      <dl
        class="column-definition-list"
      >
        <dt
          class="description__title"
        >
          Neighborhood
        </dt>
        <dd
          class="description__body"
        >
          <span>
            Tenderloin
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Built
        </dt>
        <dd
          class="description__body"
        >
          <span>
            2018
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Appliances
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Landing at The San Francisco Shipyard: Refrigerator: Bosch Stainless Steel Refrigerator Bosch - 300 Series Compact Washer and Condensation Dryer Bosch Stainless Steel Slide In Gas Range with Hood Bosch Stainless Steel Dishwasher Honeywell Thermostat
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Services Covered by HOA Dues
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Test Service
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Parking
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Landing at The San Francisco Shipyard:

            <br />
            <br />
            One parking space is included in the sale price of each unit. The parking space will be resold with the unit within the maximum resale price.

            <br />
            <br />
            <br />
            <br />
            Palisades at The San Francisco Shipyard:

            <br />
            <br />
            Two bay parking spaces in attached garage are included in the sale price of each unit. The parking space will be resold with the unit within the maximum resale price.
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Smoking Policy
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Smoking may cause problems
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Pets Policy
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            All pets are awesome!
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Unit Details
        </dt>
        <dd
          class="description__body"
        >
          <a
            href="https://www.example.com"
            target="_blank"
          >
            Download price and income limit for each unit (PDF).
          </a>
        </dd>
        <dt
          class="description__title"
        />
        <dd
          class="description__body"
        />
      </dl>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                2 Bedroom
              </strong>
              : 2 units, 1199 - 1273 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                3 Bedroom
              </strong>
              : 5 units, 1041 - 1355 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                4 Bedroom
              </strong>
              : 1 unit, 1615 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`ListingDetailsFeatures displays listing details features section when rental listing 1`] = `
<DocumentFragment>
  <li
    class="responsive-content-item "
  >
    <header
      class="detail-header "
    >
      <span
        class="detail-header__image-container"
      >
        <img
          alt=""
          class="detail-header__image "
          src="listing-features.svg"
        />
      </span>
      <hgroup
        class="detail-header__hgroup"
      >
        <h2
          class="detail-header__title"
        >
          Features
        </h2>
        <span
          class="detail-header__subtitle"
        >
          Amenities, unit details and additional fees
        </span>
        <span
          class="ui-icon ui-medium"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </hgroup>
    </header>
    <div
      class="listing-detail-panel"
    >
      <dl
        class="column-definition-list"
      >
        <dt
          class="description__title"
        >
          Neighborhood
        </dt>
        <dd
          class="description__body"
        >
          <span>
            Hayes Valley
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Built
        </dt>
        <dd
          class="description__body"
        >
          <span>
            2009
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Smoking Policy
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Non-smoking building
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Pets Policy
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Dogs are not allowed in the building.
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Property Amenities
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Laundry room, underground parking, courtyard, bike room
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Accessibility
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Elevator to all floors
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Unit Features
        </dt>
        <dd
          class="description__body"
        />
      </dl>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                2 Bedroom
              </strong>
              : 2 units, 1199 - 1273 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                3 Bedroom
              </strong>
              : 5 units, 1041 - 1355 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                4 Bedroom
              </strong>
              : 1 unit, 1615 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
      <div
        class="info-card bg-gray-100 border-0"
      >
        <div
          class="info-card__header"
        >
          <h3
            class="info-card__title"
          >
            Additional Fees
          </h3>
        </div>
        <section
          class="grid-section"
        >
          <div
            class="grid-section__header-container"
          />
          <div
            class="grid-section__inner md:grid md:grid-cols-2 md:gap-8 mb-5"
          >
            <div
              aria-label=""
              class="grid-item"
            >
              <div
                class="text-base"
              >
                Deposit
              </div>
              <div
                class="text-xl font-bold"
              >
                $2,102 - $2,355
              </div>
              <div
                class="text-sm"
              >
                or one month's rent
              </div>
              <div
                class="text-sm"
              >
                May be higher for lower credit scores
              </div>
            </div>
          </div>
        </section>
        <div
          class="info-card__columns text-xs"
        >
          <div
            class="info-card__column-2"
          >
            <span
              class="translate"
            >
              <p>
                Tenants pay for gas, electricity.

                <br />
                <br />
                For pet fees:  Cat is allowed with a $500 refundable deposit, $250 non-refundable cleaning fee and a pet addendum.
                <br />
                <br />
                <br />
                Dogs are not allowed in the building.
                <br />
                <br />
                <br />
                One parking space per unit available for $175 a month.
              </p>
            </span>
          </div>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`ListingDetailsFeatures displays listing details features section when sales listing 1`] = `
<DocumentFragment>
  <li
    class="responsive-content-item "
  >
    <header
      class="detail-header "
    >
      <span
        class="detail-header__image-container"
      >
        <img
          alt=""
          class="detail-header__image "
          src="listing-features.svg"
        />
      </span>
      <hgroup
        class="detail-header__hgroup"
      >
        <h2
          class="detail-header__title"
        >
          Features
        </h2>
        <span
          class="detail-header__subtitle"
        >
          Amenities and unit details
        </span>
        <span
          class="ui-icon ui-medium"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </hgroup>
    </header>
    <div
      class="listing-detail-panel"
    >
      <dl
        class="column-definition-list"
      >
        <dt
          class="description__title"
        >
          Neighborhood
        </dt>
        <dd
          class="description__body"
        >
          <span>
            Tenderloin
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Built
        </dt>
        <dd
          class="description__body"
        >
          <span>
            2018
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Appliances
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Landing at The San Francisco Shipyard: Refrigerator: Bosch Stainless Steel Refrigerator Bosch - 300 Series Compact Washer and Condensation Dryer Bosch Stainless Steel Slide In Gas Range with Hood Bosch Stainless Steel Dishwasher Honeywell Thermostat
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Services Covered by HOA Dues
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Test Service
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Parking
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Landing at The San Francisco Shipyard:

            <br />
            <br />
            One parking space is included in the sale price of each unit. The parking space will be resold with the unit within the maximum resale price.

            <br />
            <br />
            <br />
            <br />
            Palisades at The San Francisco Shipyard:

            <br />
            <br />
            Two bay parking spaces in attached garage are included in the sale price of each unit. The parking space will be resold with the unit within the maximum resale price.
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Smoking Policy
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            Smoking may cause problems
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Pets Policy
        </dt>
        <dd
          class="description__body translate"
        >
          <span>
            All pets are awesome!
          </span>
        </dd>
        <dt
          class="description__title"
        >
          Unit Details
        </dt>
        <dd
          class="description__body"
        >
          <span />
        </dd>
        <dt
          class="description__title"
        />
        <dd
          class="description__body"
        />
      </dl>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                2 Bedroom
              </strong>
              : 2 units, 1199 - 1273 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                3 Bedroom
              </strong>
              : 5 units, 1041 - 1355 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
      <div
        class="mb-4"
      >
        <button
          aria-expanded="false"
          class="w-full "
          data-testid="content-accordion-button"
        >
          <div
            class="flex justify-between  accordion-blue-theme__bar  "
          >
            <h3
              class="toggle-header-content"
            >
              <strong>
                4 Bedroom
              </strong>
              : 1 unit, 1615 square feet
            </h3>
            <span
              class="ui-icon ui-base flex items-center"
              data-testid="accordion-open"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
